documentation/docs/get-started/setup.mdx

```diff
@@ -97,24 +97,44 @@ sudo -u postgres createdb khoj --password
 
 ##### Local Server Setup
 - *Make sure [python](https://realpython.com/installing-python/) and [pip](https://pip.pypa.io/en/stable/installation/) are installed on your machine*
+- Check [llama-cpp-python setup](https://python.langchain.com/docs/integrations/llms/llamacpp#installation) if you hit any llama-cpp issues with the installation
 
 Run the following command in your terminal to install the Khoj backend.
 
 ```mdx-code-block
   <Tabs groupId="operating-systems">
     <TabItem value="macos" label="MacOS">
     ```shell
+# ARM/M1+ Machines
+MAKE_ARGS="-DLLAMA_METAL=on" python -m pip install khoj-assistant
+
+# Intel Machines
```

Comment:
Hmm, could we have this automatically determine OS/GPU and do this in code during setup?

Reply:
That'd have been great but not sure how we can do that via the pip install mechanism given that llama.cpp builds python wheels on the client device to handle a wider range of GPUs/machines.

If our install step becomes too complex, we could ask folks to use an installer script (e.g `curl https://khoj.dev/install.sh | bash`) but for now this mechanism of using env variables should be good enough to start with as
- the self-hosted version is anyway for folks comfortable with the terminal
- other projects our using the same setup mechanism